shared/api/network_addresses.go

```diff
@@ -0,0 +1,55 @@
+package api
+
+// Resource is a generic type used to abstract resource control features
+// support for the legacy and unified hierarchy.
+type ConsumingNetworkEntity int
+
+const (
+	// An instance consuming network resources.
+	InstanceNetConsumer ConsumingNetworkEntity = iota
+	// An external resource consuming network resources.
+	ExternalNetConsumer
+	// A network forward consuming network resources.
+	NetworkForwardNetConsumer
+	// A network load-balancer consuming network resources.
+	NetworkLoadBalancerNetConsumer
+	// A network consuming network resources.
+	NetworkNetConsumer
+)
+
+func (c ConsumingNetworkEntity) String() string {
```

Comment:
What are these used for, they dont seem to appear in the  NetworkAllocation struct fields?